Apps/phironnet/ironnet_connector.py

```diff
@@ -80,136 +81,105 @@ def __init__(self):
         try:
-            r = request_func(
-                url,
-                auth=(self._username, self._password),  # basic authentication
-                verify=self._verify_server_cert,
-                data=json.dumps(data),
-                **kwargs)
-        except Exception as e:
-            if e.message:
-                if isinstance(e.message, basestring):
-                    error_msg = UnicodeDammit(e.message).unicode_markup.encode('UTF-8')
-                else:
-                    try:
-                        error_msg = str(e.message)
-                    except:
-                        error_msg = "Unknown error occurred. Please check the asset configuration parameters."
+            if error_code in ERROR_CODE_MSG:
+                error_text = "Error Message: {0}".format(error_msg)
             else:
-                error_msg = "Unknown error occurred. Please check the asset configuration parameters."
-            self.save_progress("Error while issuing REST call - {}".format(error_msg))
-            return RetVal(
-                action_result.set_status(phantom.APP_ERROR, "Error Connecting to server. Details: {0}".format(error_msg)),
-                None)
+                error_text = "Error Code: {0}. Error Message: {1}".format(
+                    error_code, error_msg)
+        except:
+            self.debug_print("Error occurred while parsing error message")
+            error_text = PARSE_ERR_MSG
 
-        return self._process_response(r, action_result)
+        return error_text
 
     def _handle_test_connectivity(self, param):
         action_result = self.add_action_result(ActionResult(dict(param)))
 
         self.save_progress("Attempting to connect to IronAPI")
 
-        # make rest call
-        ret_val, response = self._make_post('/Login', action_result, data=None, headers=None)
+        call_status = None
 
-        if phantom.is_success(ret_val):
+        try:
+            response = self._api_instance.login(swagger_client.IronapiTypesLoginRequest(), _preload_content=False)
+            self.save_progress("Received response: Code:{}, Data:{}".format(response.status, self._handle_py_ver_compat_for_input_str(response.data)))
+            if response.status != 200:
+                call_status = phantom.APP_ERROR
+                action_result.set_status(call_status, "Fetch for Login failed. Status code was {}".format(response.status))
+            else:
+                call_status = phantom.APP_SUCCESS
+        except ApiException as e:
+            err = self._get_error_message_from_exception(e)
+            call_status = phantom.APP_ERROR
+            action_result.set_status(call_status, 'A server error has occurred: {}'.format(err))
+        except Exception as e:
+            err = self._get_error_message_from_exception(e)
+            call_status = phantom.APP_ERROR
+            action_result.set_status(call_status, 'A server error has occurred: {}'.format(err))
```

Comment:
are these supposed to print different status messages? otherwise, it doesn't seem necessary to have the `APIException` block

same applies to all the instances with code duplicated between the `ApiException` and `Exception` blocks in this file (did not want to copy the comment repeatedly)

Reply:
Fixed